firebase-common/src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -416,6 +416,7 @@ protected FirebaseApp(Context applicationContext, String name, FirebaseOptions o
     this.applicationContext = Preconditions.checkNotNull(applicationContext);
     this.name = Preconditions.checkNotEmpty(name);
     this.options = Preconditions.checkNotNull(options);
+    StartupTime startupTime = StartupTime.now();
```

Comment:
Not sure if I made this comment before but I think it's too late to record the start up time at this point. Can we instead capture it when FirebaseInitProvider initializes?

Additionally we want to make sure we only capture this value if we are "auto-initializing" i.e. if the developer manually initializes FirebaseApp we would not treat it as app starting up.

Reply:
Sure, I initially misinterpreted and thought you wanted it earlier in the constructor.